### PR TITLE
CATTY-18 WhenBackgroundChangesScript

### PR DESCRIPTION
**Pull request by aeluna**
Executes a script when the background of the current SpriteObject changes to a given background.
Changes include a new Script (WhenBackgroundChangesScript), the corresponding condition, as well as as the necessary files to visualise the script as a brick. 

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty/DataModel/Scripts/WhenBackgroundChangesScript.m [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+@implementation WhenBackgroundChangesScript
+
+- (kBrickCategoryType)category
+{
+    return kControlBrick;
+}
+
+- (NSString*)pathForLook
+{
+    return [NSString stringWithFormat:@"%@%@/%@", [self.object projectPath], kProjectImagesDirName, self.look.fileName];
```
Comment (m-herold) [inline src/Catty/DataModel/Scripts/WhenBackgroundChangesScript.m:35]: The path to the looks has changed. Please use `[look pathForScene]` instead.

File: src/CattyTests/Parser & Serialization/CatrobatLanguage0.991/XMLParserTests/Unit Tests/XMLParserTests0991.swift [hunk 2]
```diff
@@ -537,6 +537,15 @@ class XMLParserTests0991: XMLAbstractTest {
         XCTAssertTrue(whenTouchDownScript.isKind(of: WhenTouchDownScript.self), "Invalid script type")
     }
 
+    func testWhenBackgroundChangedScript() {
+        let project = self.getProjectForXML(xmlFile: "ValidProjectAllBricks0991")
```
Comment (m-herold) [inline src/CattyTests/Parser & Serialization/CatrobatLanguage0.991/XMLParserTests/Unit Tests/XMLParserTests0991.swift:541]: Can you please move this script from _ValidProjectAllBricks0991_ to _ValidProjectAllBricks0992_ and also to `XMLParserTests0992.swift`?

File: src/Catty/Condition/WhenBackgroundChangesScript+Condition.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+extension WhenBackgroundChangesScript: CBConditionProtocol {
+
+    func checkCondition(formulaInterpreter: FormulaInterpreterProtocol) -> Bool {
+        guard let backgroundObject = self.object.scene.objects().first else { return false }
+
+        if backgroundObject.spriteNode.currentLook!.isEqual(to: self.look) {
```
Comment (m-herold) [inline src/Catty/Condition/WhenBackgroundChangesScript+Condition.swift:28]: Please try to remove the force unwrapping.

**Review by m-herold — CHANGES_REQUESTED** [review]
Thanks! Looking very good already, just left a few minor suggestions.

File: src/Catty/PlayerEngine/DataModel/CBSpriteNode/CBSpriteNode.swift [hunk 4]
```diff
@@ -25,7 +25,13 @@ class CBSpriteNode: SKSpriteNode {
 
     // MARK: - Properties
     @objc var spriteObject: SpriteObject
-    @objc var currentLook: Look?
+    @objc var currentLook: Look? {
+        didSet {
+            guard let stage = self.scene as? StageProtocol else { return }
+            if !self.spriteObject.isBackground() { return }
```
Comment (m-herold) [inline src/Catty/PlayerEngine/DataModel/CBSpriteNode/CBSpriteNode.swift:31]: Please also add a test case documenting that `notifyBackgroundChange` is not called for normal objects which are not background objects.

**Review by m-herold — CHANGES_REQUESTED** [review]

File: src/Catty/DataModel/Scripts/WhenBackgroundChangesScript.m (hunk 1 above)
Comment (aeluna) [reply src/Catty/DataModel/Scripts/WhenBackgroundChangesScript.m]: Thank you for the hint! I changed it.

File: src/CattyTests/Parser & Serialization/CatrobatLanguage0.991/XMLParserTests/Unit Tests/XMLParserTests0991.swift (hunk 2 above)
Comment (aeluna) [reply src/CattyTests/Parser & Serialization/CatrobatLanguage0.991/XMLParserTests/Unit Tests/XMLParserTests0991.swift]: Sure, done. :)

File: src/Catty/Condition/WhenBackgroundChangesScript+Condition.swift (hunk 3 above)
Comment (aeluna) [reply src/Catty/Condition/WhenBackgroundChangesScript+Condition.swift]: Thank you for pointing it out! Done.

File: src/Catty/PlayerEngine/DataModel/CBSpriteNode/CBSpriteNode.swift (hunk 4 above)
Comment (aeluna) [reply src/Catty/PlayerEngine/DataModel/CBSpriteNode/CBSpriteNode.swift:31]: Thanks for the suggestion, I added a new test.

**Review by m-herold — APPROVED** [review]

**Comment by m-herold** [comment]
Thanks!

**Pull request merged**